llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -2978,6 +2984,12 @@ static bool isProfitableToSpeculate(const BranchInst *BI, bool Invert,
   return BIEndProb < Likely;
 }
 
+static bool isSafeCheapLoadStore(const Instruction &I,
+                                 const TargetTransformInfo &TTI) {
+  return (isa<LoadInst>(I) || isa<StoreInst>(I)) &&
```

Comment:
Is it safe to convert a volatile load/store into a masked load/store?

Reply:
Probably not. Checked by `not_atomic`, `not_volatile`